### ISSUE TITLE
Feature (removal) request: Remove keyboard shortcut for transpose

### ISSUE THREAD

**Issue by kendonB** [rstudio/rstudio#4549]
For years I kept finding instances of pairs of adjacent characters within my code being swapped. I thought I was either going nuts, had been hacked, or a colleague thought they were funny.

Turns out, I had been in advertently pressing Ctrl-T to try and get new scripts to appear (just like new tabs in Chrome). I had just not noticed that the code was changing once I remembered I wasn't inside Chrome.

In short, I thought it would be good to remove the default keyboard shortcut for transpose text as it's easy for people to accidentally use it.

An argument against doing so is it will temporarily make life difficult for those who are active users of the shortcut.

**Comment by gtritchie** [comment]
Are you using RStudio Desktop (and if so, on what platform)? Or RStudio Server (i.e. in a web browser)?

On macOS, the Ctrl+T for transpose is a standard thing supported by text edit fields in most applications (one of many Emacs keybindings that Mac uses). That said, I wasn't aware of it either, until now so I can see it being irritating given the overlap with New Tab.

So, I'd be fine with having no default key binding for this one, but you say it's hard to know how many people would be sad about that. So let's see what other commands and/or thumbs-ups we get.

**Comment by kevinushey** [comment]
Are you able to modify the keyboard shortcut for Transpose Letters? E.g. you could remove the <kbd>Ctrl + T</kbd> binding altogether.

**Comment by gtritchie** [comment]
It can be customized or removed, but the request is to "remove the default keyboard shortcut for transpose text".

**Comment by jmcphers** [comment]
This binding is as old as the hills and ultimately is derived from GNU Readline.

> transpose-chars (C-t)
> 
> Drag the character before the cursor forward over the character at the cursor, moving the cursor forward as well. If the insertion point is at the end of the line, then this transposes the last two characters of the line. Negative arguments have no effect.

https://tiswww.case.edu/php/chet/readline/readline.html

Maybe we could keep it working in the RStudio console, where we generally respect readline bindings, but disable it for the source editor.

**Comment by kendonB** [comment]
A fresh install seems to have this at Alt-Shift-X now. Will close.
